Dependencies installation command for Ubuntu/Linux

There is a wrong name for the Boost apt-get package in the [installation guide for Linux Developers](https://github.com/giotto-ai/giotto-tda#linux): 
~`sudo apt-get install cmake boost`~
`sudo apt-get install cmake libboost-dev`

Also when building from sources with pip, in the dependencies python-igraph installation failed for me due to,
```
    /usr/bin/ld: cannot find -lxml2
    collect2: error: ld returned 1 exit status
    make[3]: *** [Makefile:2357: libigraph.la] Error 1
```
so I think adding `lxml2-dev` to the list might also be useful.

--- a/.azure-ci/docker_scripts.sh
+++ b/.azure-ci/docker_scripts.sh
@@ -29,7 +29,7 @@ cd /io
 pip install -e ".[tests, doc]"
 
 # Test dev install with pytest and flake8
-pytest --cov gtda --no-cov --no-coverage-upload
+pytest gtda --no-cov --no-coverage-upload
 flake8 --exit-zero /io/
 
 # Uninstall giotto-tda/giotto-tda-nightly dev

--- a/.azure-ci/docker_scripts.sh
+++ b/.azure-ci/docker_scripts.sh
@@ -29,7 +29,7 @@ cd /io
 pip install -e ".[tests, doc]"
 
 # Test dev install with pytest and flake8
-pytest gtda --no-cov --no-coverage-upload
+pytest --cov . --cov-report xml
 flake8 --exit-zero /io/
 
 # Uninstall giotto-tda/giotto-tda-nightly dev

--- a/.azure-ci/docker_scripts.sh
+++ b/.azure-ci/docker_scripts.sh
@@ -2,7 +2,7 @@
 
 set -x
 
-# Upgrading pip and setuptools, TODO: Monitor status of pip versions
+# Upgrade pip and setuptools. TODO: Monitor status of pip versions
 PYTHON_PATH=$(eval find "/opt/python/*${python_ver}*" -print)
 export PATH=${PYTHON_PATH}/bin:${PATH}
 pip install --upgrade pip==19.3.1 setuptools
@@ -29,13 +29,13 @@ cd /io
 pip install -e ".[tests, doc]"
 
 # Test dev install with pytest and flake8
-pytest --cov . --cov-report xml
+pytest --cov gtda --no-cov --no-coverage-upload
 flake8 --exit-zero /io/
 
 # Uninstall giotto-tda/giotto-tda-nightly dev
 pip uninstall -y giotto-tda
 pip uninstall -y giotto-tda-nightly
 
-# Building wheels
+# Build wheels
 pip install wheel
 python setup.py sdist bdist_wheel

--- a/.azure-ci/docker_scripts.sh
+++ b/.azure-ci/docker_scripts.sh
@@ -29,7 +29,7 @@ cd /io
 pip install -e ".[tests, doc]"
 
 # Test dev install with pytest and flake8
-pytest --cov . --cov-report xml
+pytest --cov gtda --cov-report xml
 flake8 --exit-zero /io/
 
 # Uninstall giotto-tda/giotto-tda-nightly dev

--- a/.azure-ci/docker_scripts.sh
+++ b/.azure-ci/docker_scripts.sh
@@ -24,15 +24,17 @@ cd ..
 export BOOST_ROOT=/boost
 export Boost_INCLUDE_DIR=/boost/include
 
-# Install and uninstall giotto-tda dev
+# Install dev environment
 cd /io
 pip install -e ".[tests, doc]"
-pip uninstall -y giotto-tda
-pip uninstall -y giotto-tda-nightly
 
-# Testing, linting
+# Test dev install with pytest and flake8
 pytest --cov . --cov-report xml
 flake8 --exit-zero /io/
+
+# Uninstall giotto-tda/giotto-tda-nightly dev
+pip uninstall -y giotto-tda
+pip uninstall -y giotto-tda-nightly
 
 # Building wheels
 pip install wheel

--- a/README.rst
+++ b/README.rst
@@ -95,7 +95,9 @@ bug fixes can be installed by running   ::
 
 The main difference between giotto-tda-nightly and the developer installation (see the section
 on contributing, below) is that the former is shipped with pre-compiled wheels (similarly to the stable
-release) and hence does not require any C++ dependencies.
+release) and hence does not require any C++ dependencies. As the main library module is called ``gtda`` in
+both the stable and nightly versions, giotto-tda and giotto-tda-nightly should not be installed in
+the same environment.
 
 Contributing
 ------------
@@ -126,7 +128,7 @@ Most Linux systems should come with a suitable compiler pre-installed. For the o
 
 .. code-block:: bash
 
-    sudo apt-get install cmake boost
+    sudo apt-get install cmake libboost-dev
 
 if ``apt-get`` is available in your system.
 

--- a/README.rst
+++ b/README.rst
@@ -1,4 +1,4 @@
-.. image:: https://www.giotto.ai/static/vector/logo.svg
+.. image:: https://www.giotto.ai/static/vector/logo-tda.svg
    :width: 850
 
 |Version|_ |Azure-build|_ |Azure-cov|_ |Azure-test|_ |Twitter-follow|_ |Slack-join|_

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -29,7 +29,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --cov-report xml
+      pytest --cov gtda --no-coverage-upload
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -129,7 +129,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --cov-report xml
+      pytest --cov gtda --no-coverage-upload
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -393,7 +393,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --cov-report xml
+      pytest --cov gtda --no-cov --no-coverage-upload
       flake8
     failOnStderr: true
     displayName: 'Test dev install with pytest and flake8'

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -29,7 +29,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --no-coverage-upload
+      pytest gtda --no-cov --no-coverage-upload
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -80,7 +80,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --cov-report xml
+      pytest gtda --cov --cov-report xml
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -129,7 +129,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --no-coverage-upload
+      pytest gtda --no-cov --no-coverage-upload
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -279,7 +279,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --cov-report xml
+      pytest gtda --cov --cov-report xml
       flake8
     failOnStderr: true
     displayName: 'Test dev install with pytest and flake8'
@@ -393,7 +393,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest --cov gtda --no-cov --no-coverage-upload
+      pytest gtda --no-cov --no-coverage-upload
       flake8
     failOnStderr: true
     displayName: 'Test dev install with pytest and flake8'

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -29,7 +29,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest gtda --no-cov --no-coverage-upload
+      pytest --cov gtda --cov-report xml
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -129,7 +129,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest gtda --no-cov --no-coverage-upload
+      pytest --cov gtda --cov-report xml
       flake8
     failOnStderr: true
     displayName: 'Test with pytest and flake8'
@@ -279,7 +279,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest gtda --cov --cov-report xml
+      pytest --cov gtda --cov-report xml
       flake8
     failOnStderr: true
     displayName: 'Test dev install with pytest and flake8'
@@ -393,7 +393,7 @@ jobs:
     displayName: 'Install dev environment'
 
   - script: |
-      pytest gtda --no-cov --no-coverage-upload
+      pytest --cov gtda --cov-report xml
       flake8
     failOnStderr: true
     displayName: 'Test dev install with pytest and flake8'

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -66,7 +66,7 @@ jobs:
 
   - script: |
       brew update
-      brew install gcc boost
+      brew install boost
     failOnStderr: true
     displayName: 'Install gcc and boost'
 
@@ -265,7 +265,7 @@ jobs:
 
   - script: |
       brew update
-      brew install gcc boost
+      brew install boost
     failOnStderr: true
     displayName: 'Install gcc and boost'
 

--- a/gtda/utils/tests/test_validation.py
+++ b/gtda/utils/tests/test_validation.py
@@ -4,8 +4,8 @@
 import numpy as np
 import pytest
 
-from ..validation import check_diagram
-from ..validation import validate_metric_params, validate_params
+from gtda.utils.validation import (check_diagram, validate_metric_params,
+                                   validate_params)
 
 
 # Testing for validate_params

